backend/src/StdLibCliHost/Libs/Cli.fs

```diff
@@ -143,6 +145,119 @@ let fns : List<BuiltInFn> =
         | _ -> incorrectArgs ()
       sqlSpec = NotQueryable
       previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn [ "Cli" ] "executeFunction" 0
+      typeParams = []
+      parameters =
+        [ Param.make "functionName" TString ""
+          Param.make "args" (TList TString) "" ]
+      returnType =
+        TypeReference.result
+          TString
+          (TCustomType(FQName.BuiltIn(typ [ "Cli" ] "ExecutionError" 0), []))
+      description = "Executes an arbitrary Dark function"
+      fn =
+        function
+        | state, [], [ DString functionName; DList args ] ->
+          uply {
+            let err (msg : string) (metadata : List<string * string>) =
+              let metadata = metadata |> List.map (fun (k, v) -> k, DString v) |> Map
+              Dval.resultError (
+                Dval.record
+                  (FQName.BuiltIn(typ [ "Cli" ] "ExecutionError" 0))
+                  [ "msg", DString msg; "metadata", DDict metadata ]
+              )
+
+            let exnError (e : exn) : Dval =
+              let msg = Exception.getMessages e |> String.concat "\n"
+              let metadata =
+                Exception.toMetadata e |> List.map (fun (k, v) -> k, string v)
+              err msg metadata
+
+            try
+              let parts = functionName.Split('.') |> List.ofArray
+              let name = NEList.ofList "PACKAGE" parts
+              let resolver = LibParser.NameResolver.fromExecutionState state
+              let fnName =
+                LibParser.NameResolver.FnName.resolve
+                  resolver
+                  []
+                  (WT.Unresolved name)
+
+              match fnName with
+              | Ok fnName ->
+                match
+                  NEList.find Dval.isFake (NEList.ofList args.Head args.Tail)
```

Comment:
the args.Head might fail - we could do this to avoid it:
```
match fnName, args with
| Ok fnName, firstArg :: additionalArgs -> 
  match NEList.find Dval.isFake (NEList.ofList firstArg additionalArgs) with
  | ...
   ...
| _ -> incorrectArgs
`